diff --git a/build/json-c/CMakeLists.txt b/build/json-c/CMakeLists.txt
index 79038aa..d9aabfb 100644
--- a/build/json-c/CMakeLists.txt
+++ b/build/json-c/CMakeLists.txt
@@ -432,9 +432,6 @@ add_library(${PROJECT_NAME}
     ${JSON_C_SOURCES}
     ${JSON_C_HEADERS}
 )
-set_target_properties(${PROJECT_NAME} PROPERTIES
-    VERSION 5.1.0
-    SOVERSION 5)
 list(APPEND CMAKE_TARGETS ${PROJECT_NAME})
 # If json-c is used as subroject it set to target correct interface -I flags and allow
 # to build external target without extra include_directories(...)
