diff --git a/build/util-linux/libtool b/build/util-linux/libtool
index 48cea9b..818bf5e 100644
--- a/build/util-linux/libtool
+++ b/build/util-linux/libtool
@@ -9666,8 +9666,8 @@ func_mode_link ()
 
 	linux) # correct to gnu/linux during the next big refactor
 	  func_arith $current - $age
-	  major=.$func_arith_result
-	  versuffix=$major.$age.$revision
+	  major=
+	  versuffix=
 	  ;;
 
 	osf)
diff --git a/build/util-linux/Makefile b/build/util-linux/Makefile
index 0efad4d59..256164808 100644
--- a/build/util-linux/Makefile
+++ b/build/util-linux/Makefile
@@ -15362,12 +15362,9 @@ uninstall-man: uninstall-man1 uninstall-man3 uninstall-man5 \
 install-exec-hook-libuuid:
 	if test "$(usrlib_execdir)" != "$(libdir)" -a -f "$(DESTDIR)$(usrlib_execdir)/libuuid.so"; then \
 		$(MKDIR_P) $(DESTDIR)$(libdir); \
-		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
+		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so $(DESTDIR)$(libdir); \
 		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
 		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
-		(cd $(DESTDIR)$(usrlib_execdir) && \
-			rm -f libuuid.so && \
-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
 	fi
 
 uninstall-hook-libuuid:
@@ -15377,12 +15374,9 @@ uninstall-hook-libuuid:
 install-exec-hook-libblkid:
 	if test "$(usrlib_execdir)" != "$(libdir)" -a -f "$(DESTDIR)$(usrlib_execdir)/libblkid.so"; then \
 		$(MKDIR_P) $(DESTDIR)$(libdir); \
-		mv $(DESTDIR)$(usrlib_execdir)/libblkid.so.* $(DESTDIR)$(libdir); \
+		mv $(DESTDIR)$(usrlib_execdir)/libblkid.so $(DESTDIR)$(libdir); \
 		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libblkid.so); \
 		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
-		(cd $(DESTDIR)$(usrlib_execdir) && \
-			rm -f libblkid.so && \
-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libblkid.so); \
 	fi
 
 uninstall-hook-libblkid:
@@ -15392,12 +15386,9 @@ uninstall-hook-libblkid:
 install-exec-hook-libmount:
 	if test "$(usrlib_execdir)" != "$(libdir)" -a -f "$(DESTDIR)$(usrlib_execdir)/libmount.so"; then \
 		$(MKDIR_P) $(DESTDIR)$(libdir); \
-		mv $(DESTDIR)$(usrlib_execdir)/libmount.so.* $(DESTDIR)$(libdir); \
+		mv $(DESTDIR)$(usrlib_execdir)/libmount.so $(DESTDIR)$(libdir); \
 		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libmount.so); \
 		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
-		(cd $(DESTDIR)$(usrlib_execdir) && \
-			rm -f libmount.so && \
-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libmount.so); \
 	fi
 
 uninstall-hook-libmount:
@@ -15407,12 +15398,9 @@ uninstall-hook-libmount:
 install-exec-hook-libsmartcols:
 	if test "$(usrlib_execdir)" != "$(libdir)" -a -f "$(DESTDIR)$(usrlib_execdir)/libsmartcols.so"; then \
 		$(MKDIR_P) $(DESTDIR)$(libdir); \
-		mv $(DESTDIR)$(usrlib_execdir)/libsmartcols.so.* $(DESTDIR)$(libdir); \
+		mv $(DESTDIR)$(usrlib_execdir)/libsmartcols.so $(DESTDIR)$(libdir); \
 		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libsmartcols.so); \
 		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
-		(cd $(DESTDIR)$(usrlib_execdir) && \
-			rm -f libsmartcols.so && \
-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libsmartcols.so); \
 	fi
 
 uninstall-hook-libsmartcols:
@@ -15422,12 +15410,9 @@ uninstall-hook-libsmartcols:
 install-exec-hook-libfdisk:
 	if test "$(usrlib_execdir)" != "$(libdir)" -a -f "$(DESTDIR)$(usrlib_execdir)/libfdisk.so"; then \
 		$(MKDIR_P) $(DESTDIR)$(libdir); \
-		mv $(DESTDIR)$(usrlib_execdir)/libfdisk.so.* $(DESTDIR)$(libdir); \
+		mv $(DESTDIR)$(usrlib_execdir)/libfdisk.so $(DESTDIR)$(libdir); \
 		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libfdisk.so); \
 		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
-		(cd $(DESTDIR)$(usrlib_execdir) && \
-			rm -f libfdisk.so && \
-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libfdisk.so); \
 	fi
 
 uninstall-hook-libfdisk:
