diff --git a/build/zlib/configure b/build/zlib/configure
index e974d1f..9c61de8 100755
--- a/build/zlib/configure
+++ b/build/zlib/configure
@@ -208,7 +208,7 @@ if test "$gcc" -eq 1 && ($cc -c $test.c) >> configure.log 2>&1; then
   fi
   case "$uname" in
   Linux* | linux* | GNU | GNU/* | solaris*)
-        LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"} ;;
+        LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so,--version-script,${SRCDIR}zlib.map"} ;;
   *BSD | *bsd* | DragonFly)
         LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"}
         LDCONFIG="ldconfig -m" ;;
diff --git a/build/zlib/Makefile.in b/build/zlib/Makefile.in
index 5a77949..9d123b7 100644
--- a/build/zlib/Makefile.in
+++ b/build/zlib/Makefile.in
@@ -32,9 +32,7 @@ CPP=$(CC) -E
 
 STATICLIB=libz.a
 SHAREDLIB=libz.so
-SHAREDLIBV=libz.so.1.2.11
-SHAREDLIBM=libz.so.1
-LIBS=$(STATICLIB) $(SHAREDLIBV)
+LIBS=$(STATICLIB)
 
 AR=ar
 ARFLAGS=rc
@@ -278,11 +276,8 @@ gzwrite.lo: $(SRCDIR)gzwrite.c
 	-@mv objs/gzwrite.o $@
 
 
-placebo $(SHAREDLIBV): $(PIC_OBJS) libz.a
+placebo $(SHAREDLIB): $(PIC_OBJS) libz.a
 	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS) $(LDSHAREDLIBC) $(LDFLAGS)
-	rm -f $(SHAREDLIB) $(SHAREDLIBM)
-	ln -s $@ $(SHAREDLIB)
-	ln -s $@ $(SHAREDLIBM)
 	-@rmdir objs
 
 example$(EXE): example.o $(STATICLIB)
@@ -291,11 +286,11 @@ example$(EXE): example.o $(STATICLIB)
 minigzip$(EXE): minigzip.o $(STATICLIB)
 	$(CC) $(CFLAGS) -o $@ minigzip.o $(TEST_LDFLAGS)
 
-examplesh$(EXE): example.o $(SHAREDLIBV)
-	$(CC) $(CFLAGS) -o $@ example.o -L. $(SHAREDLIBV)
+examplesh$(EXE): example.o $(SHAREDLIB)
+	$(CC) $(CFLAGS) -o $@ example.o -L. $(SHAREDLIB)
 
-minigzipsh$(EXE): minigzip.o $(SHAREDLIBV)
-	$(CC) $(CFLAGS) -o $@ minigzip.o -L. $(SHAREDLIBV)
+minigzipsh$(EXE): minigzip.o $(SHAREDLIB)
+	$(CC) $(CFLAGS) -o $@ minigzip.o -L. $(SHAREDLIB)
 
 example64$(EXE): example64.o $(STATICLIB)
 	$(CC) $(CFLAGS) -o $@ example64.o $(TEST_LDFLAGS)
@@ -313,15 +308,12 @@ install-libs: $(LIBS)
 	cp $(STATICLIB) $(DESTDIR)$(libdir)
 	chmod 644 $(DESTDIR)$(libdir)/$(STATICLIB)
 	-@($(RANLIB) $(DESTDIR)$(libdir)/libz.a || true) >/dev/null 2>&1
-	-@if test -n "$(SHAREDLIBV)"; then \
-	  rm -f $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBV); \
-	  cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir); \
-	  echo "cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)"; \
-	  chmod 755 $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBV); \
-	  echo "chmod 755 $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBV)"; \
-	  rm -f $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBM); \
-	  ln -s $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB); \
-	  ln -s $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBM); \
+	-@if test -n "$(SHAREDLIB)"; then \
+	  rm -f $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB); \
+	  cp $(SHAREDLIB) $(DESTDIR)$(sharedlibdir); \
+	  echo "cp $(SHAREDLIB) $(DESTDIR)$(sharedlibdir)"; \
+	  chmod 755 $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB); \
+	  echo "chmod 755 $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB)"; \
 	  ($(LDCONFIG) || true)  >/dev/null 2>&1; \
 	fi
 	rm -f $(DESTDIR)$(man3dir)/zlib.3
@@ -342,8 +335,8 @@ install: install-libs
 uninstall:
 	cd $(DESTDIR)$(includedir) && rm -f zlib.h zconf.h
 	cd $(DESTDIR)$(libdir) && rm -f libz.a; \
-	if test -n "$(SHAREDLIBV)" -a -f $(SHAREDLIBV); then \
-	  rm -f $(SHAREDLIBV) $(SHAREDLIB) $(SHAREDLIBM); \
+	if test -n "$(SHAREDLIB)" -a -f $(SHAREDLIB); then \
+	  rm -f $(SHAREDLIB) $(SHAREDLIB) $(SHAREDLIB); \
 	fi
 	cd $(DESTDIR)$(man3dir) && rm -f zlib.3
 	cd $(DESTDIR)$(pkgconfigdir) && rm -f zlib.pc
